include/components/config.hpp

```diff
@@ -238,23 +235,22 @@ class config {
   /**
    * Dereference value reference
    */
-  template <typename T>
-  T dereference(const string& section, const string& key, const string& var, const T& fallback) const {
+  string dereference(const string& section, const string& key, const string& var) const {
```

Comment:
Since this is no longer templated, we can move it (and all other dereference methods) into the cpp file, should speed up compilation a bit.

Reply:
Implementations moved to cpp file. I also moved `set` and `has` implementations.